backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlTypeUtils.java

```diff
@@ -79,16 +76,19 @@ public static void applyValidationAndComputed(List<Column> columns, Row row) {
     }
   }
 
-  private static void applyAutoid(Column c, Row row) {
+  private static void applyAutoId(Column c, Row row) {
     if (row.isNull(c.getName(), c.getPrimitiveColumnType())) {
+      SnowFlakeIdGenerator idGenerator = SnowFlakeIdGenerator.getInstance();
       // do we use a template containing ${mg_autoid} for pre/postfixing ?
       if (c.getComputed() != null) {
         row.set(
             c.getName(),
-            c.getComputed().replace(Constants.COMPUTED_AUTOID_TOKEN, idGenerator.generateId()));
+            c.getComputed()
+                .replace(
+                    Constants.COMPUTED_AUTOID_TOKEN, idGenerator.generateId(c.getTableName())));
```

Comment:
Good question, I tested this the only side affect is that it will mess up the sorting. Since this is a key feature of the identifier we probably don't want this. Making the algorithm configurable is the best solution?

Reply:
the keys should not be conflicting. I would like to ask data managers about this because ideally we would simply move to new scheme while keeping old data using old ids, and new data using new ids